app/src/main/java/cm/aptoide/pt/home/HomePresenter.java

```diff
@@ -160,6 +193,8 @@ public HomePresenter(HomeView view, Home home, Scheduler viewScheduler, CrashRep
         .observeOn(viewScheduler)
         .doOnNext(created -> view.showLoading())
         .flatMapSingle(created -> loadBundles())
+        .filter(bundlesModel -> !hasAppNextHighlightedAd(bundlesModel))
+        .flatMapSingle(__ -> loadAppNextAd("ads-highlighted"))
```

Comment:
In the future we need to change this. The loadbundles should emit each time a new bundle loads. (highlighted, appnext, recommends) and not block the view while waiting.
This should be only a quick fix now since it requires a lot of effort doing the refactor at the moment.